helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementService.java

```diff
@@ -771,4 +797,86 @@ protected Map<String, Boolean> getInstanceHealthStatus(String clusterId, String
 
     return healthStatus;
   }
+
+  public static class MaintenanceManagementServiceBuilder {
+    private ConfigAccessor _configAccessor;
+    private boolean _skipZKRead;
+    private String _namespace;
+    private ZKHelixDataAccessor _dataAccessor;
+    private CustomRestClient _customRestClient;
+    private boolean _continueOnFailure;
+    private Set<StoppableCheck.Category> _skipHealthCheckCategories = Collections.emptySet();
+    private List<HealthCheck> _stoppableHealthCheckList = Collections.emptyList();
+
+    public MaintenanceManagementServiceBuilder setConfigAccessor(ConfigAccessor configAccessor) {
+      _configAccessor = configAccessor;
+      return this;
+    }
+
+    public MaintenanceManagementServiceBuilder setSkipZKRead(boolean skipZKRead) {
+      _skipZKRead = skipZKRead;
+      return this;
+    }
+
+    public MaintenanceManagementServiceBuilder setNamespace(String namespace) {
+      _namespace = namespace;
+      return this;
+    }
+
+    public MaintenanceManagementServiceBuilder setDataAccessor(
+        ZKHelixDataAccessor dataAccessor) {
+      _dataAccessor = dataAccessor;
+      return this;
+    }
+
+    public MaintenanceManagementServiceBuilder setCustomRestClient(
+        CustomRestClient customRestClient) {
+      _customRestClient = customRestClient;
+      return this;
+    }
+
+    public MaintenanceManagementServiceBuilder setContinueOnFailure(boolean continueOnFailure) {
+      _continueOnFailure = continueOnFailure;
+      return this;
+    }
+
+    public MaintenanceManagementServiceBuilder setSkipHealthCheckCategories(
+        Set<StoppableCheck.Category> skipHealthCheckCategories) {
+      _skipHealthCheckCategories = skipHealthCheckCategories;
+      return this;
+    }
+
+    public MaintenanceManagementServiceBuilder setStoppableHealthCheckList(
+        List<HealthCheck> stoppableHealthCheckList) {
+      _stoppableHealthCheckList = stoppableHealthCheckList;
```

Comment:
When user pass [ALL], we could consolidate a list with all checks.

Reply:
I think a better logic would be
* If the stoppableCheckList is given, Helix should perform the given checks.
* If the stoppableCheckList == null, Helix will perform all existing Helix own checks. This approach preserves backward compatibility, ensuring that existing users receive the full suite of checks they anticipate by default.

In this way, users don't need to pass [ALL] to indicate they want to all stoppable checks.